oxidation/libparsec/crates/client_types/src/local_device_file.rs

```diff
@@ -94,12 +96,44 @@ impl DeviceFile {
             .map_err(|_| LocalDeviceError::Access(key_file_path.to_path_buf()))
     }
 
-    fn load(key_file_path: &Path) -> LocalDeviceResult<Self> {
+    pub fn load(key_file_path: &Path) -> LocalDeviceResult<Self> {
         let data = std::fs::read(key_file_path)
             .map_err(|_| LocalDeviceError::Access(key_file_path.to_path_buf()))?;
 
-        rmp_serde::from_slice::<DeviceFile>(&data)
-            .map_err(|_| LocalDeviceError::Deserialization(key_file_path.to_path_buf()))
+        let result = rmp_serde::from_slice::<DeviceFile>(&data)
+            .map_err(|_| LocalDeviceError::Deserialization(key_file_path.to_path_buf()));
+
+        // In case of failure try to load a legacy_device and convert it to a
+        // regular device file
+        if result.is_err() {
+            Self::load_legacy_device_file(key_file_path, &data)
```

Comment:
You can use https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else

Reply:
Strange `clippy` don't say anything about that :thinking:

Reply:
Done.